packages/react-querybuilder/src/utils/formatQuery/formatQuery.test.ts

```diff
@@ -630,10 +631,9 @@ const params_named = {
+const mongoQueryString = `{"$and":[{},{"firstName":null},{"lastName":{"$ne":null}},{"firstName":{"$in":["Test","This"]}},{"lastName":{"$nin":["Test","This"]}},{"firstName":{"$in":[]}},{"firstName":{"$gte":"Test","$lte":"This"}},{"firstName":{"$gte":"Test","$lte":"This"}},{"lastName":{"$lte":"Test","$gte":"This"}},{"firstName":{"$gte":"","$lte":""}},{"firstName":{"$gte":"","$lte":""}},{"age":{"$gte":12,"$lte":14}},{"age":"26"},{"isMusician":true},{"email":{"$regex":"@"}},{"email":{"$regex":"^ab"}},{"email":{"$regex":"com$"}},{"hello":{"$not":{"$regex":"com"}}},{"job":{"$not":{"$regex":"^Man"}}},{"job":{"$not":{"$regex":"ger$"}}},{"$or":[{"job":"Sales Executive"},{"job":{"$in":[]}},{"job":{"$gte":"just one value","$lte":""}}]}]}`;
 const mongoQueryStringForValueSourceField =
-  '{"$and":[{"firstName":null},{"lastName":{"$ne":null}},{"$where":"[this.middleName,this.lastName].includes(this.firstName)"},{"$where":"![this.middleName,this.lastName].includes(this.lastName)"},{"$and":[{"$expr":{"$gte":["$firstName","$middleName"]}},{"$expr":{"$lte":["$firstName","$lastName"]}}]},{"$and":[{"$expr":{"$gte":["$firstName","$middleName"]}},{"$expr":{"$lte":["$firstName","$lastName"]}}]},{"$or":[{"$expr":{"$lt":["$lastName","$middleName"]}},{"$expr":{"$gt":["$lastName","$lastName"]}}]},{"$expr":{"$eq":["$age","$iq"]}},{"$expr":{"$eq":["$isMusician","$isCreative"]}},{"$where":"this.email.includes(this.atSign)"},{"$where":"this.email.startsWith(this.name)"},{"$where":"this.email.endsWith(this.dotCom)"},{"$where":"!this.hello.includes(this.dotCom)"},{"$where":"!this.job.startsWith(this.noJob)"},{"$where":"!this.job.endsWith(this.noJob)"},{"$or":[{"$expr":{"$eq":["$job","$executiveJobName"]}}]}]}';
+  '{"$and":[{},{"firstName":null},{"lastName":{"$ne":null}},{"firstName":{"$in":["middleName","lastName"]}},{"lastName":{"$nin":["middleName","lastName"]}},{"firstName":{"$in":[]}},{"firstName":{"$gte":"middleName","$lte":"lastName"}},{"firstName":{"$gte":"middleName","$lte":"lastName"}},{"lastName":{"$lte":"middleName","$gte":"lastName"}},{"firstName":{"$gte":"","$lte":""}},{"firstName":{"$gte":"","$lte":""}},{"age":"iq"},{"isMusician":"isCreative"},{"$where":"this.email.includes(this.atSign)"},{"$where":"this.email.startsWith(this.name)"},{"$where":"this.email.endsWith(this.dotCom)"},{"$where":"!this.hello.includes(this.dotCom)"},{"$where":"!this.job.startsWith(this.noJob)"},{"$where":"!this.job.endsWith(this.noJob)"},{"$or":[{"job":"executiveJobName"}]}]}';
```

Comment:
What's with the addition of the empty rule object (first element of the `$and` array)?

Reply:
`{
      field: 'invalid',
      operator: 'invalid',
      value: '',
      valueSource: 'field',
    }`
It's for a rule with an invalid operator, how should we handle this, i was thinking of showing it in the query, indicating that it's an invalid rule. The user can then still remove the invalid rule or make the rule valid

Reply:
You should use [validation](https://react-querybuilder.js.org/docs/api/validation) to style invalid rules within the `<QueryBuilder />` component instead of producing invalid output with `formatQuery`.